onnx/defs/schema.h

```diff
@@ -655,14 +662,21 @@ class OpSchemaRegistry final {
   }
 };
 
-#define ONNX_OPERATOR_SCHEMA(name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(__COUNTER__, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(Counter, name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)                 \
+// ONNX_OPERATOR_SCHEMA is implemented either with static registration or by exposing
+// a factory method.  
+#ifndef ONNX_NON_STATIC_SCHEMA_INITIALIZATION
```

Comment:
I'm nervous about adding this kind of option, similar with the one we added for ONNX_ML. it makes the code diverge and different partners focus on each option. It's generally hard to maintain. For such divergence, either all partners agree on same direction, or put the specific stuff in local code base.

Reply:
I agree that having different patterns is not ideal.  However, not all vendors may be able to leverage the code in its current state, but the code contains a lot of necessary information.  To answer the question above @bddppq, control of the timing of initialization is important to manage potential startup cost for the compilation unit.  As we are looking at this, the code can be brought into many different processes, some where deferring this cost is important.

Reply:
For non-framework scenarios, when this code is loaded as part of an ONNX library, it can be consumed by numerous processes.  We do a lot of work to eliminate this overhead, delay initialization until the library is used, etc...  The dynamic initializer makes this code hard to consume.  Not sure if providing benchmarks changes this.

Reply:
@shschaefer I'm asking benchmark numbers mainly because I think the overhead of these static initialization should be negligible.

Reply:
I measured that registering schema consumes 4 milliseconds and increases memory consumption by 1.1 MB.  This is very non-negligible for a single component, especially if it may be loaded into many processes.